enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,501 @@
+        foo: bar
+    - CRD
+      resource: dnsrecords
+    - service:
+      namespace: foo
+      publishInternal: "true"
+status:
+  conditions:
+  - lastTransitionTime: "2020-08-20T23:01:33Z"
+    reason: Valid
+    status: "True"
+    type: Admitted
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    status: "True"
+    type: Available
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    message: The deployment has Available status condition set to True
+    reason: DeploymentAvailable
+    status: "False"
+    type: DeploymentDegraded
+  provider:
+    type: AWS
+    credentials: my-secret-reference
+```
+
+Additional `spec` and/or `status` fields may be introduced based on requirements.
+
+By exposing broad spec fields such as `zones` and `provider`, ExternalDNS operator users would have the ability to
+enable ExternalDNS for multiple providers across multiple DNS zones. In general, a single instance of the ExternalDNS Operator
+CRD would correlate to a single [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) of ExternalDNS.
+Note that multiple instances of ExternalDNS (operands) would be necessary when a common domain is shared across DNS zones,
+or when using multiple DNS Providers. See ExternalDNS' upstream
+[AWS public/private zone tutorial docs](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/public-private-route53.md)
+as an example. In this example, it may be acceptable to run one ExternalDNS instance for a given public zone, and one ExternalDNS instance for a given private zone,
+via the same deployment (See the [Open Questions](#open-questions) section).
+
+The `zones` field contains a list of DNS zone IDs. On OpenShift, the ExternalDNS Operator could read
+the cluster DNS config resource and extract the available public/private zone IDs. Note that a zone lookup may need
+to be performed by the ExternalDNS Operator in order to determine the ID of any zones that are specified
+via a resource tag in the cluster DNS config resource. The ExternalDNS operator will then take the provided zone IDs
+and map them to the given provider's complete hosted zone ID string.
+See the ExternalDNS deployment example loosely based off of this ExternalDNS Operator CRD instance in [Implementation Details](#implementation-details).
+
+On OpenShift, the ExternalDNS Operator could create a `credentialsrequest.cloudcredential.openshift.io` resource so that
+the cluster administrator would not have to worry about providing cloud credentials by hand when creating an ExternalDNS Operator resource.
+ExternalDNS requires different permissions for each cloud provider, as documented in the
+[tutorial section of the ExternalDNS docs](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials).
+In general, ExternalDNS needs proper permissions to view, update, and delete DNS resource records in the relevant DNS zones.
+
+Support for OpenShift clusters running AWS that
```

Comment:
Can we just say "Support for AWS clusters that use STS"?  The CredentialsRequest API is OpenShift-specific, but given that we'll need some way to provide credentials on vanilla Kubernetes, does support for STS need to be OpenShift-specific?

Reply:
Right, we can. Fixed.